codegen/gql_code_builder/lib/src/required_vars_constructor.dart

```diff
@@ -0,0 +1,60 @@
+import "package:built_collection/built_collection.dart";
+import "package:code_builder/code_builder.dart";
+import "package:gql/ast.dart";
+import "package:gql_code_builder/source.dart";
+import "package:gql_code_builder/src/common.dart";
+
+Constructor builtCreateConstructor({
+  required String name,
+  required Iterable<Method> getters,
+  required SourceNode schemaSource,
+}) {
+  final className = builtClassName(name);
+
+  /// Getters that are excluded from the create constructor
+  /// because they are system getters that are not part of the data
+  final excludedGettersFromCreateConstructor = ["G__typename"];
+
+  // Filter out excluded getters
+  final filteredGetters = getters.where(
+    (g) => !excludedGettersFromCreateConstructor.contains(g.name),
+  );
+
+  // Create parameters for the factory constructor
+  final factoryParameters = filteredGetters.map(
+    (g) => Parameter(
+      (b) => b
+        ..name = g.name!
+        ..named = true
+        ..required = true
+        ..type = g.returns,
+    ),
+  );
+
+  final assignments = filteredGetters.map((g) {
+    final typeDefinitionNode = getTypeDefinitionNode(
+        schemaSource.document, g.returns!.symbol!.replaceFirst("G", ""));
+    final isBuiltType = typeDefinitionNode is InputObjectTypeDefinitionNode ||
```

Comment:
The `is` type testing seemed to be the best way forward here, curious to learn wether there are easier solutions?

Reply:
Additionally, I couldn't come up with any other edge-cases other than InputTypes and Scalars, but please lmk if there are any others